lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -364,6 +364,24 @@ void PatchEntryPointMutate::gatherUserDataUsage(Module *module) {
           // index.
           descriptorTable.resize(std::max(descriptorTable.size(), size_t(set + 1)));
           descriptorTable[set].users.push_back(call);
+
+          if (m_pipelineState->getOptions().resourceLayoutScheme == ResourceLayoutScheme::Indirect) {
+            // In indirect scheme, The order of resources is like this:
+            // 1. one user data entry for vertex buffer
+            // 2. one user data entry for the push constant buffer
+            // 3. descriptor set index for each set
+            // If the type is pushconst, the index is set 0, others are set + 1.
```

Comment:
This type of comment should be somewhere more discoverable.  If I am reading `DescBuilder.cpp` I might want this information.  Could you place this with the definition of `ResourceLayoutScheme` in include/vkgcDefs.h.